backend/benefit/messages/permissions.py

```diff
@@ -0,0 +1,18 @@
+from django.conf import settings
+from django.utils.translation import gettext_lazy as _
+from rest_framework import permissions
+from users.utils import get_company_from_request
+
+
+class IsMessageAuthor(permissions.BasePermission):
+    message = _("You don't have permission to change this message")
+
+    def has_object_permission(self, request, view, obj):
+        if (
+            settings.DISABLE_AUTHENTICATION
+            or request.method in permissions.SAFE_METHODS
+        ):
+            return True
+        if request.user.is_handler():
+            return obj.sender == request.user
```

Comment:
Maybe we should confirm with Joonas if this restriction applies? Should the handlers be able to delete/edit each other's messages?

Reply:
So Jonathan confirm that we do not allow update, only delete is allowed. I update this part to follow the specification. Relevant unittests are also updated